pkg/placement/plugins/addon/addon.go

```diff
@@ -82,7 +82,7 @@ func (c *AddOn) Score(ctx context.Context, placement *clusterapiv1beta1.Placemen
 		// get AddOnPlacementScores CR with resourceName
 		addOnScores, err := c.handle.ScoreLister().AddOnPlacementScores(namespace).Get(c.resourceName)
 		if err != nil {
-			klog.Warningf("Getting AddOnPlacementScores failed: %s", err)
+			klog.FromContext(ctx).Info("Failed to get AddOnPlacementScores", "error", err)
```

Comment:
How about adding the logger to schedulerHandler, so that each plugin can use the logger directly, don't need to init it each time by plugin itself. 
https://github.com/haoqing0110/OCM/commit/deffa73b2d94fe61cab62173df7016e1e6a0e54f express my idea.

Reply:
A different opinion, to me add the logger to schedulerHandler feels like we still hold a global logger but in another format.

Recap the idea of contextual log https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/3077-contextual-logging/README.md , we want:

```
Log calls that were already converted to structured logging need to be updated to use a logger that gets passed into the function in one of two ways:

as explicit parameter
attached to a context.Context
```

Reply:
Hey @haoqing0110 thanks, it is a great idea. I have updated it as per this. Please take a look

Reply:
@xuezhaojun It is a valid point, if we want to configure at a function level.

Reply:
@nitishchauhan0022 @xuezhaojun 's suggestion makes sense I think, checking the Kubernetes scheduler, they init the logger inside plugin https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/noderesources/resource_allocation.go#L53

Reply:
Ok, i will revert back to previous changes